src/main/java/org/sasanlabs/service/vulnerability/xss/reflected/XSSInImgTagAttribute.java

```diff
@@ -168,18 +183,22 @@ public ResponseEntity<String> getVulnerablePayloadLevel6(
             variant = Variant.SECURE,
             htmlTemplate = "LEVEL_1/XSS")
     public ResponseEntity<String> getVulnerablePayloadLevelSecure(
-            @RequestParam Map<String, String> queryParams) {
+            @RequestParam(PARAMETER_NAME) String imageLocation) {
         String vulnerablePayloadWithPlaceHolder = "<img src=\"%s\" width=\"400\" height=\"300\"/>";
-        StringBuilder payload = new StringBuilder();
 
-        for (Map.Entry<String, String> map : queryParams.entrySet()) {
-            if (allowedValues.contains(map.getValue())) {
-                payload.append(
-                        String.format(
-                                vulnerablePayloadWithPlaceHolder,
-                                StringEscapeUtils.escapeHtml4(map.getValue())));
-            }
+        if ((imageLocation.startsWith(IMAGE_RESOURCE_PATH)
+                        && imageLocation.endsWith(FILE_EXTENSION))
+                || allowedValues.contains(imageLocation)) {
```

Comment:
```
|| allowedValues.contains(imageLocation)
```
 this check is also not required. and also please run `./gradlew spotlessApply`

Reply:
But before you explicitly asked me to add that part. I guess, you lost me a bit, sorry.

Reply:
aah no, i think there is a confusion, i asked for level6 not level7 as level7 is added by you.

Reply:
you can even left this condition as well. please run `./gradlew spotlessApply` and we can merge this PR. thanks